mantidproject/mantid
# Fix sporadic unit test failure of ConvertSpiceDataToRealData

What was happening here is that the `DownloadInstrumentTest` creates a temporary folder in the instrument folder. If it's run at the same time as the `LoadInstrument` algorithm (which is run several times by `ConvertSpiceDataToRealData`) it can pick up the temporary test folder when parsing the IDF folder contents.

`Poco::File().isFile()` throws an exception if the C `stat` function fails for some reason (probably because the directory was subsequently removed). On the other hand `Poco::Path().isFile()` just returns `false` if it's not a file and we safely ignore the testing directory.

**To test:**
 - Build `DataHandlingTest` and `MDAlgorithmsTest`
 - This test only fails sporadically. I found I could get it to fail fairly quickly by running the following:
```
for i in `seq 10`; do ctest -R "(ConvertSpice|DownloadInstrument)" -j8 --schedule-random --output-on-failure; done;
``` 
 - Once you can reproduce, apply the fix then try and get it to fail again.
 - Check there are no other unit test failures.

Fixes #20520 

**Release Notes** 
*Does not need to be in the release notes.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
